src/Components/Server/src/Circuits/CircuitHost.cs

```diff
@@ -77,29 +78,38 @@ public static void SetCurrentCircuitHost(CircuitHost circuitHost)
 
         public Circuit Circuit { get; }
 
-        public IClientProxy Client { get; }
+        public IClientProxy Client { get; set; }
 
         public IJSRuntime JSRuntime { get; }
 
         public RemoteRenderer Renderer { get; }
 
         public RendererRegistry RendererRegistry { get; }
 
+        public IList<ComponentDescriptor> Descriptors { get; }
+
         public IServiceProvider Services { get; }
 
+        public Task<IEnumerable<string>> PrerrenderComponentAsync(Type componentType, ParameterCollection parameters)
+        {
+            return _dispatcher.InvokeAsync(async () =>
+            {
+                Renderer.StartPrerrender();
```

Comment:
Is the idea that, later, we'll add an `EndPrerender` to switch it back into interactive mode once the client reconnects?

Reply:
Yes, my idea is that we would have a TCS, and a _displayUpdating task that is by default Task.CompletedTask. By default we send the renders one after the other right away, but when we prerrender we replace _displayUpdating with the task from the TCS, so that we get to queue all the initial renders.

Upon reconnection we call TCS.SetResult and all the prerender batches get sent back to the client and rendering resumes as normal.